src/append/rolling_file/mod.rs

```diff
@@ -182,7 +180,16 @@ impl Append for RollingFileAppender {
 
         // TODO(eas): Idea: make this optionally return a future, and if so, we initialize a queue for
         // data that comes in while we are processing the file rotation.
-        self.policy.process(&mut file)
+
+        //first, rotate
+        self.policy.process(&mut file)?;
+
+        //second, write
+        let writer_file = self.get_writer(&mut writer)?;
+        self.encoder.encode(writer_file, record)?;
+        writer_file.flush()?;
```

Comment:
moving this here is a net change for the size rollers, we need to preserve those semantics for both types of trigger

Reply:
Sorry, I didn't catch you. Could you please explain it in more detail. Thanks!

Reply:
I change the sort to guarantee the triger will be trigered at the right time without small overflow.